tests/resource_response_test.py

```diff
@@ -87,6 +88,23 @@ def test_cancelled_returns_false_if_called_before_cancel(self):
         self.assertFalse(self.future.cancelled())
 
 
+class HandleResponseErrorsTest(unittest.TestCase):
+
+    def test_pass_response_and_request(self):
+        request = Mock()
+        response = Mock(text='{"foo": "bar"}')
+        response.json.return_value = {'foo': 'bar'}
+        error = Mock(
+            args=('400 Bad Request',), response=response, request=request
+        )
+        try:
+            handle_response_errors(error)
+        except HTTPError as e:
+            self.assertEqual(e.response.json(), {'foo': 'bar'})
+            self.assertEqual(e.response, response)
+            self.assertEqual(e.request, request)
```

Comment:
I am not sure if there is a test already that '400 Bad Request' is contained in the `e`'s args, will be good to add it if not present already.